Amazon.IonDotnet/Internals/Text/SystemTextReader.cs

```diff
@@ -526,24 +526,30 @@ public override string[] GetTypeAnnotations()
             for (int index = 0; index < _annotations.Count; index++)
             {
                 SymbolToken symbolToken = _annotations[index];
-                if (symbolToken.Text is null && symbolToken.ImportLocation != default)
+                if (symbolToken.Text is null)
                 {
-                    ISymbolTable symtab = GetSymbolTable();
-
-                    string text = symtab.FindKnownSymbol(symbolToken.ImportLocation.Sid);
-                    if (text == null)
+                    if (symbolToken.ImportLocation == default)
                     {
-                        throw new UnknownSymbolException(symbolToken.ImportLocation.Sid);
+                        throw new UnknownSymbolException(symbolToken.Sid);
                     }
+                    else if (symbolToken.ImportLocation != default)
```

Comment:
Last thing: this is just an `else {}` now.

Reply:
Updated to else.